packages/flutter/lib/src/material/data_table.dart

```diff
@@ -311,6 +313,20 @@ class DataTable extends StatelessWidget {
   /// row is selectable.
   final ValueSetter<bool> onSelectAll;
 
+  /// The horizontal padding between the edges of the table and the content
+  /// in the first and last cells of each row.
+  ///
+  /// When a checkbox is displayed, it is also the padding between the checkbox
+  /// the content in the first data column.
+  ///
+  /// This value is optional and defaults to 24.0.
+  final double tablePadding;
```

Comment:
One thing that jumps out at me is that this is called "padding", but there are other paddings in the Flutter API that are of type `EdgeInsetsGeometry` instead of `double` (i.e. in [Container](https://api.flutter.dev/flutter/widgets/Container-class.html)).

I'm not sure if you can do anything about that, but I just wanted to point it out.  I see you create an `EdgeInsetsDirectional` below, but maybe it doesn't make sense to accept one here.  Maybe there are other "paddings" in the framework that are doubles too, I don't know.

Reply:
The main reason I made this choice was because currently, DataTable calculates height separately and the [other PR](https://github.com/flutter/flutter/pull/33535/) accounts for dynamic heights. Having an `EdgeInsetsGeometry` would start making things a tad more confusing, since you can now have a height, as well as adjust vertical padding. 

That could be a welcome change though, but at the time when I was writing this, I felt like it could be confusing and I'm actually not sure how the interaction between height and vertical padding would play out, but I can certainly take another look once I remind myself what the code does.

Reply:
Sounds good to me, just bringing attention to it.

Reply:
Maybe we should give this a name that doesn't imply that it's an EdgeInsets (etc) value, maybe horizontalMargin?

It might also make sense to use a different parameter to define the space between the checkbox and the first column.

Reply:
I've updated the name to `horizontalMargin`.

As for the checkbox margins, I've considered that option and it is certainly doable. I mainly worry about a potentially confusing API. This is what the API would look like given that a `horizontalCheckboxMargin` was provided given the current way that `DataTable` is implemented.

```
  /// The horizontal margin between the edges of the data columns and the
  /// content in the first and last cells of each row.
  ///
  /// When a checkbox is not displayed, the full margin value applies 
  /// between the edges of the data table and the first data column.
  ///
  /// When a checkbox is displayed, the margin between the checkbox
  /// and the content in the first data column will be half the value 
  /// (horizontalMargin / 2.0). The distance between the last data column and 
  /// the edge of the table will be the full value.
  ///
  /// This value defaults to 24.0 to adhere to the Material Design specifications.
  final double horizontalMargin;

  /// The horizontal margin of the checkbox.
  ///
  /// The distance between the edge of the table will be the full value,
  /// whereas the distance between checkbox and the first data column will be
  /// half the provided value.
  ///
  /// This value defaults to 24.0 to adhere to the Material Design specifications.
  final double horizontalCheckboxMargin;
```

This is mainly because of the way that the horizontal margin of the first data column is calculated. It uses `horizontalMargin`, and explaining that it would be half the value given a checkbox column could be confusing to users. The tradeoff for implementing it would be that users would definitely have more control over customizing the margins.

Reply:
Discussed with Hans offline, decided it is best to move forward without `horizontalCheckboxMargin`, since that would be confusing to developers.